tests/dryrun_test.go

```diff
@@ -684,20 +675,12 @@ var _ = Describe("[sig-compute]Dry-Run requests", decorators.SigCompute, func()
 			Expect(err).ToNot(HaveOccurred())
 
 			By("Make a Dry-Run request to update a VM Restore")
-			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
-				restore, err = virtClient.VirtualMachineRestore(restore.Namespace).Get(context.Background(), restore.Name, metav1.GetOptions{})
-				if err != nil {
-					return err
-				}
-
-				restore.Labels = map[string]string{
-					"key": "42",
-				}
+			restore, err := virtClient.VirtualMachineRestore(restore.Namespace).Get(context.Background(), restore.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
 
-				opts := metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}}
-				_, err = virtClient.VirtualMachineRestore(restore.Namespace).Update(context.Background(), restore, opts)
-				return err
-			})
+			patch := []byte(`{"metadata":{"labels":{"key":"42"}}}`)
```

Comment:
Can you please use `patch.New(...).GeneratePayload()` function instead?

Reply:
It is only for `JSONPatchType`, and we should use here `MergePatchType`
which is easier, and already wide used over kubevirt
(see please https://github.com/kubevirt/kubevirt/pull/12429/files?diff=split&w=0#r1689371880)

Reply:
Okay, thanks. Isn't `(`{"metadata":{"labels":{"key":"42"}}}`)` is a JSON content?

Reply:
`MergePatchType` is json as well, RFC 7386
this is the input style for it.
`JSONPatchType` need other style
all in the link at the bottom of https://github.com/kubevirt/kubevirt/pull/12429#discussion_r1689508211
Please see the link there https://erosb.github.io/post/json-patch-vs-merge-patch/ it explains nicely on the input differences and the other differences such as complexity and such

Reply:
Thanks :)